src/i965_device_info.c

```diff
@@ -440,6 +441,7 @@ static struct hw_codec_info kbl_hw_codec_info = {
     .render_init = gen9_render_init,
     .post_processing_context_init = gen9_post_processing_context_init,
     .max_resolution = gen9_max_resolution,
+    .preinit_hw_codec = gen9_hw_codec_preinit,
```

Comment:
Agree. In fact the issue not only exists for VDENC, but also exits for the other codecs.(H264/HEVC/VP8/VP9).
The problem is related with that vaCreateConfig/vaCreateContext doesn't check whether the input rate_control is supported. 
The driver needs to be enhanced to improve the robustness.